src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -40,23 +40,29 @@ public void init() {
   }
 
   /**
-   * Responds with a JSON representation of {@link Message} data for a specific user. Responds with
-   * an empty array if the user is not provided.
+   * Responds with a JSON representation of {@link Message} data for logged-in user and another user. Responds with
+   * an empty array if the user is not logged in or another user is not provided.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     response.setContentType("application/json");
 
-    String user = request.getParameter("user");
-
-    if (user == null || user.equals("")) {
+    // Get logged-in user and other user
+    String loggedInUser = null;
+    String otherUser = request.getParameter("user");
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      loggedInUser = userService.getCurrentUser().toString();
+    }
+    
+    if (loggedInUser == null || loggedInUser.equals("") || otherUser == null || otherUser.equals("")) {
       // Request is invalid, return empty array
       response.getWriter().println("[]");
```

Comment:
In error cases, it is typical to return a differently-structured response that conveys relevant error information.

For example:

      {
          "error": {
             "message": "Missing required query parameter 'user'"
          }
      }

For how Google handles errors in its own APIs, see:
https://cloud.google.com/apis/design/errors